components/script/dom/htmloptionscollection.rs

```diff
@@ -0,0 +1,186 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::ElementBinding::ElementMethods;
+use dom::bindings::codegen::Bindings::HTMLCollectionBinding::HTMLCollectionMethods;
+use dom::bindings::codegen::Bindings::HTMLOptionsCollectionBinding;
+use dom::bindings::codegen::Bindings::HTMLOptionsCollectionBinding::HTMLOptionsCollectionMethods;
+use dom::bindings::codegen::Bindings::NodeBinding::NodeBinding::NodeMethods;
+use dom::bindings::codegen::UnionTypes::{HTMLOptionElementOrHTMLOptGroupElement, HTMLElementOrLong};
+use dom::bindings::error::{Error, ErrorResult};
+use dom::bindings::global::GlobalRef;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::js::{Root, RootedReference};
+use dom::bindings::reflector::reflect_dom_object;
+use dom::bindings::str::DOMString;
+use dom::element::Element;
+use dom::htmlcollection::{CollectionFilter, HTMLCollection};
+use dom::htmloptionelement::HTMLOptionElement;
+use dom::node::{document_from_node, Node};
+use dom::window::Window;
+
+#[dom_struct]
+pub struct HTMLOptionsCollection {
+    collection: HTMLCollection,
+}
+
+impl HTMLOptionsCollection {
+    fn new_inherited(root: &Node, filter: Box<CollectionFilter + 'static>) -> HTMLOptionsCollection {
+        HTMLOptionsCollection {
+            collection: HTMLCollection::new_inherited(root, filter),
+        }
+    }
+
+    pub fn new(window: &Window, root: &Node, filter: Box<CollectionFilter + 'static>)
+        -> Root<HTMLOptionsCollection>
+    {
+        reflect_dom_object(box HTMLOptionsCollection::new_inherited(root, filter),
+                           GlobalRef::Window(window),
+                           HTMLOptionsCollectionBinding::Wrap)
+    }
+
+    fn add_new_elements(&self, count: u32) -> ErrorResult {
+        let root = self.collection.root_node();
+        let document = document_from_node(root.r());
+
+        for _ in 0..count {
+            let element = HTMLOptionElement::new(atom!("option"), None, document.r());
+            let node = element.upcast::<Node>();
+            try!(root.AppendChild(node).map(|_| ()));
```

Comment:
@jdm, @KiChjang  I guess I was wrong, being wrapped inside try! is works without map. It doesn't in direct return cases, but do work in this one.